todo_app/data/trello_items.py

```diff
@@ -0,0 +1,124 @@
+# Everything s assuming that there are no errors anywhere.
+
+import requests
+import os
+
+api_key = os.getenv("APIKey")
+api_token = os.getenv("APIToken")
+board_id = os.getenv("BoardId")
+
+list_of_cards = []
+
+def get_idlist(name):
+    """
+    Fetches all lists from the trello and find the one that matches the name supplied.
+
+    Returns:
+        string: The idList id for the specified name.
+    """
+
+    response = requests.get(f"https://api.trello.com/1/boards/{board_id}/lists?key={api_key}&token={api_token}&cards=open&card_fields=id,name")
+
+    idlist = ""
+
+    #now look for the to_do list
+    response_json = response.json()
+
+    for trello_list in response_json:
+        if trello_list["name"] == name:
+            idlist = trello_list["id"]
+            break
+
+    return idlist
+
+def get_cards():
+    """
+    Fetches all cards from the trello which are from the 'To Do' or 'Done' lists.
+
+    Returns:
+        list: The list of all relevant.
+    """
+    response = requests.get(f"https://api.trello.com/1/boards/{board_id}/lists?key={api_key}&token={api_token}&cards=open&card_fields=id,name,idList")
+
+    list_of_cards.clear()
+
+    #now look for the to_do list
+    response_json = response.json()
+
+    for trello_list in response_json:
+        list_name = trello_list["name"]
+        cards = trello_list["cards"]
+
+        for card in cards:
+            # only interested in 'To Do' and 'Done' lists
+            if list_name == "To Do" or list_name == "Done":
+                item = {
+                    "title": card["name"],
+                    "id": card["id"],
+                    "status": list_name
+                }
+                list_of_cards.append(item)
+    
+    return list_of_cards
+
+def add_card(title):
+    """
+    Adds a new card with the specified title to the 'To Do' list.
+
+    Args:
+        title: The title of the card.
+
+    Returns:
+        item: The API response.
+    """
+
+    todo_list = get_idlist("To Do")
+
+    # set up the card details as required
+    details = "name=" + title
+    details = details + "&pos=top"
+    details = details + "&desc=This is a new card"
+
+    response = requests.post(f"https://api.trello.com/1/cards?idList={todo_list}&key={api_key}&token={api_token}&{details}")
```

Comment:
Though this works, I'm always wary of constructing these strings ourselves - plus query strings are a little non-standard on POST requests. What might be nicer is passing it through as parameters/data, something like:
```suggestion
    body = {
        'name': title,
        'pos': 'top',
        'desc': 'This is a new card',
        'key': api_key,
        'token': api_token,
        'idList': todo_list
    }
    
    response = requests.post("https://api.trello.com/1/cards", data = body)
```